src/mame/drivers/nes_clone.cpp

```diff
@@ -397,6 +458,142 @@ void nes_clone_state::nes_clone_pal(machine_config &config)
 	m_maincpu->add_route(ALL_OUTPUTS, "mono", 0.50);
 }
 
+/**************************************************
+ Dancing Expert specifics
+**************************************************/
+
+void nes_clone_dancexpt_state::machine_start()
+{
+	nes_clone_state::machine_start();
+	m_ppu->space(AS_PROGRAM).install_read_bank(0x0000, 0x1fff, m_bank_gfx);
+	m_bank_gfx->set_base(memregion("gfx1")->base());
+
+	m_nt_ram.resize(0x800);
+
+	for (int i = 0; i < 4; i++)
+		m_nametables[i]->configure_entries(0, 2, &m_nt_ram[0], 0x400);
+
+	save_item(NAME(m_nt_ram));
+
+	m_nametables[0]->set_entry(0);
+	m_nametables[1]->set_entry(1);
+	m_nametables[2]->set_entry(0);
+	m_nametables[3]->set_entry(1);
+
+	m_ppu->space(AS_PROGRAM).install_readwrite_handler(0x2000, 0x3eff, read8sm_delegate(*this, FUNC(nes_clone_dancexpt_state::nametable_r)), write8sm_delegate(*this, FUNC(nes_clone_dancexpt_state::nametable_w)));
+}
+
+
+uint8_t nes_clone_dancexpt_state::in0_r()
+{
+	// polled frequently during game, music data? from some other device?
+	return 0xfe; // bit 0x01 exits the song?
+}
+
+uint8_t nes_clone_dancexpt_state::in1_r()
+{
+	// read directly, not through shifter here
+	return m_io1->read();
+}
+
+void nes_clone_dancexpt_state::machine_reset()
+{
+	nes_clone_state::machine_reset();
+	m_5000_val = 0x6;
+}
+
+uint8_t nes_clone_dancexpt_state::bankrom_r(offs_t offset)
+{
+//	printf("%s: bankrom_r %04x\n", machine().describe_context().c_str(), offset);
+	return m_mainrom[((m_5000_val & 0x7) * 0x4000) + offset];
+}
```

Comment:
Wouldn’t it be more efficient and clear to use configured banking for this?  You’ve got eight banks with fixed stride.